File: frontend/src/types/proveedor.ts
export type CategoriaProveedor = 'materiales' | 'mano_de_obra' | 'servicios' | 'equipamiento' | 'otros'
export type EstadoProveedor = 'activo' | 'inactivo' | 'suspendido'
export type EspecialidadContratista = 'albanileria' | 'plomeria' | 'electricidad' | 'acabados' | 'carpinteria' | 'pintura' | 'soldadura' | 'vidrieria' | 'jardineria' | 'impermeabilizacion' | 'herreria' | 'general' | 'otro'
export type TipoAccion = 'computo' | 'acopio' | 'cuenta_corriente' | 'pedido' | 'entrega' | 'pago' | 'ajuste' | 'otro'
export type EstadoAccion = 'pendiente' | 'en_proceso' | 'completado' | 'cancelado'

export interface Proveedor {
  id: string
  razon_social: string
  nombre_comercial: string | null
  rfc: string | null
  categoria: CategoriaProveedor
  estado: EstadoProveedor
  especialidad: EspecialidadContratista | null
  contacto_principal: string
  email: string
  telefono: string | null
  telefono_alternativo: string | null
  direccion: string | null
  ciudad: string | null
  estado_geo: string | null
  codigo_postal: string | null
  sitio_web: string | null
  banco: string | null
  cuenta_bancaria: string | null
  clabe: string | null
  numero_licencia: string | null
  tiene_seguro: boolean
  aseguradora: string | null
  numero_poliza: string | null
  vigencia_seguro: string | null
  condiciones_pago: string | null
  dias_credito: number
  descuento_pronto_pago: number
  tiempo_entrega_dias: number | null
  calificacion: number
  total_pagado: number
  total_pendiente: number
  trabajos_completados: number
  ultima_actividad: string | null
  preferido: boolean
  notas: string | null
  activo: boolean
  created_at: string
  updated_at: string
}

export interface ProveedorCreate {
  razon_social: string
  nombre_comercial?: string
  rfc?: string
  categoria: CategoriaProveedor
  estado?: EstadoProveedor
  especialidad?: EspecialidadContratista
  contacto_principal: string
  email: string
  telefono?: string
  telefono_alternativo?: string
  direccion?: string
  ciudad?: string
  estado_geo?: string
  codigo_postal?: string
  sitio_web?: string
  banco?: string
  cuenta_bancaria?: string
  clabe?: string
  numero_licencia?: string
  tiene_seguro?: boolean
  aseguradora?: string
  numero_poliza?: string
  vigencia_seguro?: string | null
  condiciones_pago?: string
  dias_credito?: number
  descuento_pronto_pago?: number
  tiempo_entrega_dias?: number | null
  calificacion?: number
  preferido?: boolean
  notas?: string
}

export interface ProveedorUpdate extends Partial<ProveedorCreate> {
  total_pagado?: number
  total_pendiente?: number
  trabajos_completados?: number
  ultima_actividad?: string
}

// Adjuntos
export interface Adjunto {
  id: string
  nombre: string
  url: string
  tipo_archivo: string | null
  created_at: string
}

export interface AdjuntoCreate {
  nombre: string
  url: string
  tipo_archivo?: string
}

// Acciones de Proveedor
export interface AccionProveedor {
  id: string
  proveedor_id: string
  tipo: TipoAccion
  descripcion: string
  monto: number | null
  fecha: string
  estado: EstadoAccion
  referencia: string | null
  notas: string | null
  adjuntos: Adjunto[]
  created_at: string
  updated_at: string
}

export interface AccionProveedorCreate {
  tipo: TipoAccion
  descripcion: string
  monto?: number
  fecha: string
  estado?: EstadoAccion
  referencia?: string
  notas?: string
}

export interface AccionProveedorUpdate extends Partial<AccionProveedorCreate> {}

// ==================== LABELS Y COLORES PARA UI ====================

export const categoriaLabels: Record<CategoriaProveedor, string> = {
  materiales: 'Materiales',
  mano_de_obra: 'Mano de Obra',
  servicios: 'Servicios',
  equipamiento: 'Equipamiento',
  otros: 'Otros',
}

export const categoriaColors: Record<CategoriaProveedor, string> = {
  materiales: 'info',
  mano_de_obra: 'orange',
  servicios: 'purple',
  equipamiento: 'cyan',
  otros: 'secondary',
}

export const estadoProveedorLabels: Record<EstadoProveedor, string> = {
  activo: 'Activo',
  inactivo: 'Inactivo',
  suspendido: 'Suspendido',
}

export const estadoProveedorColors: Record<EstadoProveedor, string> = {
  activo: 'success',
  inactivo: 'secondary',
  suspendido: 'error',
}

export const especialidadLabels: Record<EspecialidadContratista, string> = {
  albanileria: 'Albañilería',
  plomeria: 'Plomería',
  electricidad: 'Electricidad',
  acabados: 'Acabados',
  carpinteria: 'Carpintería',
  pintura: 'Pintura',
  soldadura: 'Soldadura',
  vidrieria: 'Vidriería',
  jardineria: 'Jardinería',
  impermeabilizacion: 'Impermeabilización',
  herreria: 'Herrería',
  general: 'General',
  otro: 'Otro',
}

export const tipoAccionLabels: Record<TipoAccion, string> = {
  computo: 'Cómputo',
  acopio: 'Acopio',
  cuenta_corriente: 'Cuenta Corriente',
  pedido: 'Pedido',
  entrega: 'Entrega',
  pago: 'Pago',
  ajuste: 'Ajuste',
  otro: 'Otro',
}

export const tipoAccionColors: Record<TipoAccion, string> = {
  computo: 'info',
  acopio: 'purple',
  cuenta_corriente: 'cyan',
  pedido: 'warning',
  entrega: 'success',
  pago: 'success',
  ajuste: 'orange',
  otro: 'secondary',
}

export const estadoAccionLabels: Record<EstadoAccion, string> = {
  pendiente: 'Pendiente',
  en_proceso: 'En Proceso',
  completado: 'Completado',
  cancelado: 'Cancelado',
}

export const estadoAccionColors: Record<EstadoAccion, string> = {
  pendiente: 'secondary',
  en_proceso: 'info',
  completado: 'success',
  cancelado: 'error',
}

File: backend/app/schemas/proveedor.py
from pydantic import BaseModel, Field
from datetime import datetime, date
from typing import Optional
from app.models.proveedor import CategoriaProveedor, TipoAccion, EstadoAccion, EstadoProveedor, EspecialidadContratista


# ==================== PROVEEDOR ====================

class ProveedorCreate(BaseModel):
    razon_social: str = Field(..., min_length=1, max_length=255)
    nombre_comercial: Optional[str] = None
    rfc: Optional[str] = None

    categoria: CategoriaProveedor = CategoriaProveedor.MATERIALES
    estado: EstadoProveedor = EstadoProveedor.ACTIVO
    especialidad: Optional[EspecialidadContratista] = None

    contacto_principal: str = Field(..., min_length=1, max_length=100)
    email: str = Field(..., min_length=1)
    telefono: Optional[str] = None
    telefono_alternativo: Optional[str] = None

    direccion: Optional[str] = None
    ciudad: Optional[str] = None
    estado_geo: Optional[str] = None
    codigo_postal: Optional[str] = None
    sitio_web: Optional[str] = None

    banco: Optional[str] = None
    cuenta_bancaria: Optional[str] = None
    clabe: Optional[str] = None

    # Licencia y seguros
    numero_licencia: Optional[str] = None
    tiene_seguro: bool = False
    aseguradora: Optional[str] = None
    numero_poliza: Optional[str] = None
    vigencia_seguro: Optional[date] = None

    condiciones_pago: Optional[str] = None
    dias_credito: int = 30
    descuento_pronto_pago: float = 0
    tiempo_entrega_dias: Optional[int] = None

    calificacion: int = Field(default=0, ge=0, le=5)

    preferido: bool = False
    notas: Optional[str] = None


class ProveedorUpdate(BaseModel):
    razon_social: Optional[str] = None
    nombre_comercial: Optional[str] = None
    rfc: Optional[str] = None
    categoria: Optional[CategoriaProveedor] = None
    estado: Optional[EstadoProveedor] = None
    especialidad: Optional[EspecialidadContratista] = None
    contacto_principal: Optional[str] = None
    email: Optional[str] = None
    telefono: Optional[str] = None
    telefono_alternativo: Optional[str] = None
    direccion: Optional[str] = None
    ciudad: Optional[str] = None
    estado_geo: Optional[str] = None
    codigo_postal: Optional[str] = None
    sitio_web: Optional[str] = None
    banco: Optional[str] = None
    cuenta_bancaria: Optional[str] = None
    clabe: Optional[str] = None
    numero_licencia: Optional[str] = None
    tiene_seguro: Optional[bool] = None
    aseguradora: Optional[str] = None
    numero_poliza: Optional[str] = None
    vigencia_seguro: Optional[date] = None
    condiciones_pago: Optional[str] = None
    dias_credito: Optional[int] = None
    descuento_pronto_pago: Optional[float] = None
    tiempo_entrega_dias: Optional[int] = None
    calificacion: Optional[int] = Field(None, ge=0, le=5)
    total_pagado: Optional[float] = None
    total_pendiente: Optional[float] = None
    trabajos_completados: Optional[int] = None
    ultima_actividad: Optional[date] = None
    preferido: Optional[bool] = None
    notas: Optional[str] = None


class ProveedorResponse(BaseModel):
    id: str
    razon_social: str
    nombre_comercial: Optional[str] = None
    rfc: Optional[str] = None
    categoria: CategoriaProveedor
    estado: EstadoProveedor
    especialidad: Optional[EspecialidadContratista] = None
    contacto_principal: str
    email: str
    telefono: Optional[str] = None
    telefono_alternativo: Optional[str] = None
    direccion: Optional[str] = None
    ciudad: Optional[str] = None
    estado_geo: Optional[str] = None
    codigo_postal: Optional[str] = None
    sitio_web: Optional[str] = None
    banco: Optional[str] = None
    cuenta_bancaria: Optional[str] = None
    clabe: Optional[str] = None
    numero_licencia: Optional[str] = None
    tiene_seguro: bool
    aseguradora: Optional[str] = None
    numero_poliza: Optional[str] = None
    vigencia_seguro: Optional[date] = None
    condiciones_pago: Optional[str] = None
    dias_credito: int
    descuento_pronto_pago: float
    tiempo_entrega_dias: Optional[int] = None
    calificacion: int
    total_pagado: float
    total_pendiente: float
    trabajos_completados: int
    ultima_actividad: Optional[date] = None
    preferido: bool
    notas: Optional[str] = None
    activo: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ProveedorListResponse(BaseModel):
    items: list[ProveedorResponse]
    total: int
    skip: int
    limit: int


# ==================== ADJUNTO ====================

class AdjuntoCreate(BaseModel):
    nombre: str
    url: str
    tipo_archivo: Optional[str] = None


class AdjuntoResponse(BaseModel):
    id: str
    nombre: str
    url: str
    tipo_archivo: Optional[str] = None
    created_at: datetime

    model_config = {"from_attributes": True}


# ==================== ACCION PROVEEDOR ====================

class AccionProveedorCreate(BaseModel):
    tipo: TipoAccion
    descripcion: str = Field(..., min_length=1, max_length=500)
    monto: Optional[float] = None
    fecha: date
    estado: EstadoAccion = EstadoAccion.PENDIENTE
    referencia: Optional[str] = None
    notas: Optional[str] = None


class AccionProveedorUpdate(BaseModel):
    tipo: Optional[TipoAccion] = None
    descripcion: Optional[str] = None
    monto: Optional[float] = None
    fecha: Optional[date] = None
    estado: Optional[EstadoAccion] = None
    referencia: Optional[str] = None
    notas: Optional[str] = None


class AccionProveedorResponse(BaseModel):
    id: str
    proveedor_id: str
    tipo: TipoAccion
    descripcion: str
    monto: Optional[float] = None
    fecha: date
    estado: EstadoAccion
    referencia: Optional[str] = None
    notas: Optional[str] = None
    adjuntos: list[AdjuntoResponse] = []
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class AccionProveedorListResponse(BaseModel):
    items: list[AccionProveedorResponse]
    total: int
    skip: int
    limit: int
